metadetect/metadetect.py

```diff
@@ -12,37 +16,62 @@
 logger = logging.getLogger(__name__)
 
 
-def do_metadetect(config, mbobs, rng):
-    """
-    meta-detect on the multi-band observations.  For parameters, see
-    docs on the Metadetect class
+def do_metadetect(config, mbobs, rng, nonshear_mbobs=None):
+    """Run metadetect on the multi-band observations.
+
+    Parameters
+    ----------
+    config: dict
+        Configuration dictionary. Possible entries are
+
+            metacal
+            weight
+            model
+            flux
+
+    mbobs: ngmix.MultiBandObsList
+        We will do detection and measurements on these images
+    rng: numpy.random.RandomState
+        Random number generator
+    nonshear_mbobs: ngmix.MultiBandObsList, optional
+        If not None and 'flux' is given in the config, then this mbobs will
+        be sheared and have flux measurements made at the detected positions in
+        the `mbobs`.
+
+    Returns
+    -------
+    res: dict
+        The fitting data keyed on the shear component.
     """
-    md = Metadetect(config, mbobs, rng)
+    md = Metadetect(config, mbobs, rng, nonshear_mbobs=nonshear_mbobs)
     md.go()
     return md.result
 
 
 class Metadetect(dict):
-    """
-    meta-detect on the multi-band observations.
+    """Metadetect fitter for multi-band observations.
 
     Parameters
     ----------
     config: dict
         Configuration dictionary. Possible entries are
 
             metacal
-            weight (if calculating weighted moments)
-            max (if running a max like fitter)
-            fofs (if running MOF)
-            mof (if running MOF)
+            weight
+            model
```

Comment:
explain that different models work differently?  wmom works by band and averages things, gauss does simultaneous fit, etc.